pkg/ebpf/events_pipeline.go

```diff
@@ -281,35 +281,34 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 	return out, errc
 }
 
-// computePolicies iterates through the policies that do the filtering in user space, checking whether an event should be considered.
-// If it should not, it sets the respective offset to 0.
-// Finally it returns computed policies and the names of the policies that matched.
-func (t *Tracee) computePolicies(event *trace.Event) (uint64, []string) {
+// computePoliciesMatches iterates through the policies that do the filtering in user space.
+// If the filter doesn't match, the policy bit is cleared from the matched policies bitmap.
+// Finally, the filtered bitmap is stored in the event and returned.
+func (t *Tracee) computePoliciesMatches(event *trace.Event) uint64 {
 	eventID := events.ID(event.EventID)
-	matchedPolicies := event.MatchedPolicies
-	matchedPoliciesNames := []string{}
+	bitmap := event.MatchedPolicies
 
 	for p := range t.config.Policies.Map() {
 		bitOffset := uint(p.ID)
 
 		// Events submitted with matching policies.
 		// The policy must have its bit cleared when it does not match.
-		if !utils.HasBit(matchedPolicies, bitOffset) {
+		if !utils.HasBit(bitmap, bitOffset) {
 			continue
 		}
 
```

Comment:
If the event was not chosen by this policy, you can remove the matchedPolicies bit of this policy and continue (like you did before, but also clear the bit of the policy before continue)